packages/zendesk-adapter/src/change_validators/custom_statuses_enabled.ts

```diff
@@ -60,6 +60,75 @@ const areCustomStatusesEnabled = async (
   return customStatusesEnabled.enabled
 }
 
+const createErrorsForCustomStatusTypes = (
+  changes: readonly Change<ChangeDataType>[]
+): ChangeError[] =>
+  changes
+    .filter(
+      change =>
+        getChangeData(change).elemID.typeName === CUSTOM_STATUS_TYPE_NAME
+    )
+    .map(getChangeData)
+    .map(instance => ({
+      elemID: instance.elemID,
+      severity: 'Error',
+      message: 'Custom statuses are not enabled.',
+      detailedMessage:
+        'Cannot deploy custom statuses when they are not enabled in the Zendesk account.',
+    }))
+
+type ChildField = {
+  // eslint-disable-next-line camelcase
+  required_on_statuses: {
+    // eslint-disable-next-line camelcase
+    custom_statuses?: ReferenceExpression[]
+  }
+}
+
+type Condition = {
+  // eslint-disable-next-line camelcase
+  child_fields: ChildField[]
+}
+
+type TicketValue = {
+  // eslint-disable-next-line camelcase
+  agent_conditions?: Condition[]
+  // eslint-disable-next-line camelcase
+  end_user_conditions?: Condition[]
+}
+
+const hasConditionWithCustomStatuses = (conditions: Condition[]): boolean =>
+  conditions.some((condition: Condition) =>
+    condition.child_fields.some(
+      field => (field.required_on_statuses.custom_statuses?.length ?? 0) > 0
+    ))
+
+const isTicketFormWithCustomStatus = (change: Change<ChangeDataType>): boolean => {
+  const data = getChangeData(change)
+  if (data.elemID.typeName !== TICKET_FORM_TYPE_NAME) return false
```

Comment:
please don't use shorthand in the code - 
```suggestion
  if (data.elemID.typeName !== TICKET_FORM_TYPE_NAME) {
    return false
  }
```
(and same below)

Reply:
Done 👍